File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int cnt = 0;
        for (int[] c : commands) {
            int start = c[0] - 1;
            int end = c[1];
            int find = c[2] - 1;
            
            List<Integer> arr = new ArrayList<>();
            
            for (int i = start;i<end;i++) {
                arr.add(array[i]);
            }
            
            Collections.sort(arr);
            
            answer[cnt++] = arr.get(find);
        }
        
        
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (vector<int> command : commands) {
        vector<int> temp;
        int i = command[0] - 1;
        int j = command[1] - 1;
        int k = command[2] - 1;
        
        
        for (int start = i; start<=j;start++){
            temp.push_back(array[start]);
        }
        sort(temp.begin(), temp.end());
        
        answer.push_back(temp[k]);
    }
    
    
    return answer;
}